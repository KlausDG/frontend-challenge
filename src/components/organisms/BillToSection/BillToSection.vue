<script setup lang="ts">
import { Card, Stack } from '@/components/atoms'
import { IconText } from '@/components/molecules'
import type { Address } from '@/models'

defineProps<{
  data: Address
}>()
</script>

<template>
  <section>
    <Card title="Bill to" :subtitle="data.name">
      <Stack direction="column" :spacing="4">
        <IconText icon="map-marker" :text="data.address" />
        <IconText icon="user" :text="data.contact.name" />
        <IconText icon="envelope" :text="data.contact.email" />
        <IconText icon="phone" :text="data.contact.phone" />
        <IconText icon="print" :text="data.contact.fax" />
      </Stack>
    </Card>
  </section>
</template>
