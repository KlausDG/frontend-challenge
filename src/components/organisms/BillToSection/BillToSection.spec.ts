import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import BillToSection from './BillToSection.vue'

const CardMock = {
  name: 'Card',
  props: ['title', 'subtitle'],
  template: '<div class="mock-card"><slot /></div>',
}

const StackMock = {
  name: 'Stack',
  props: ['direction', 'spacing'],
  template: '<div class="mock-stack"><slot /></div>',
}

const IconTextMock = {
  name: 'IconText',
  props: ['icon', 'text'],
  template: '<div class="mock-icon-text">{{ icon }}:{{ text }}</div>',
}

describe('BillToSection.vue', () => {
  const addressData = {
    name: 'John Doe',
    address: '123 Main St',
    label: 'teste',
    code: '123',
    contact: {
      name: 'Jane Doe',
      email: 'jane@example.com',
      phone: '123456789',
      fax: '987654321',
    },
  }

  const wrapper = mount(BillToSection, {
    props: { data: addressData },
    global: {
      components: {
        Card: CardMock,
        Stack: StackMock,
        IconText: IconTextMock,
      },
    },
  })

  it('renders Card with correct title and subtitle', () => {
    const card = wrapper.findComponent(CardMock)
    expect(card.exists()).toBe(true)
    expect(card.props('title')).toBe('Bill to')
    expect(card.props('subtitle')).toBe(addressData.name)
  })

  it('renders Stack inside Card', () => {
    const stack = wrapper.findComponent(StackMock)
    expect(stack.exists()).toBe(true)
    expect(stack.props('direction')).toBe('column')
    expect(stack.props('spacing')).toBe(16)
  })

  it('renders all IconText components with correct props', () => {
    const iconTexts = wrapper.findAllComponents(IconTextMock)
    expect(iconTexts).toHaveLength(5)

    expect(iconTexts[0].props()).toEqual({ icon: 'map-marker', text: addressData.address })
    expect(iconTexts[1].props()).toEqual({ icon: 'user', text: addressData.contact.name })
    expect(iconTexts[2].props()).toEqual({ icon: 'envelope', text: addressData.contact.email })
    expect(iconTexts[3].props()).toEqual({ icon: 'phone', text: addressData.contact.phone })
    expect(iconTexts[4].props()).toEqual({ icon: 'print', text: addressData.contact.fax })
  })

  it('matches the component snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
