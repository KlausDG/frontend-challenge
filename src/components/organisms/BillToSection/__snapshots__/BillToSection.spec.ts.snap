// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BillTo.vue > matches the component snapshot 1`] = `
"<section>
  <div data-v-69f42c1a="" data-v-9bb7b3b9="" class="stack__container card__container" style="flex-direction: column; gap: 16px;">
    <div data-v-69f42c1a="" data-v-9bb7b3b9="" class="stack__container" style="flex-direction: column; gap: 4px;">
      <p data-v-9bb7b3b9="" class="typography _caption_d0cbd2 _text_d0cbd2">Bill to</p>
      <p data-v-9bb7b3b9="" class="typography _semibold_d0cbd2 _text_d0cbd2">John Doe</p>
    </div>
    <div data-v-69f42c1a="" class="stack__container" style="flex-direction: column; gap: 4px;">
      <div data-v-70c01fa4="" class="icon-text__container">
        <unicon data-v-70c01fa4="" name="map-marker"></unicon>
        <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">123 Main St</p>
      </div>
      <div data-v-70c01fa4="" class="icon-text__container">
        <unicon data-v-70c01fa4="" name="user"></unicon>
        <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">Jane Doe</p>
      </div>
      <div data-v-70c01fa4="" class="icon-text__container">
        <unicon data-v-70c01fa4="" name="envelope"></unicon>
        <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">jane@example.com</p>
      </div>
      <div data-v-70c01fa4="" class="icon-text__container">
        <unicon data-v-70c01fa4="" name="phone"></unicon>
        <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">123456789</p>
      </div>
      <div data-v-70c01fa4="" class="icon-text__container">
        <unicon data-v-70c01fa4="" name="print"></unicon>
        <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">987654321</p>
      </div>
    </div>
  </div>
</section>"
`;
